doc/source/development/roadmap.rst

```diff
@@ -0,0 +1,187 @@
+.. _roadmap:
+
+=======
+Roadmap
+=======
+
+This page provides an overview of the major themes in pandas' development. Implementation
+of these goals may be hastened with dedicated funding or interest from contributors.
+
+An item being on the roadmap does not mean that pandas will *necessarily* happen, even
+with unlimited funding. During the implementation period we may discover issues
+preventing the adoption of the feature.
+
+See :ref:`roadmap.evolution` for proposing changes to this document.
+
+Extensibility
+-------------
+
+Pandas :ref:`extending.extension-types` allow for extending NumPy types with custom
+data types and array storage. Pandas uses extension types internally, and provides
+an interface for 3rd-party libraries to define their own custom data types.
+
+Many parts of pandas still unintentionally convert data to a NumPy array.
+These problems are especially pronounced for nested data.
+
+We'd like to improve the handling of extension arrays throughout the library,
+making their behavior more consistent with the handling of NumPy arrays. We'll do this
+by cleaning up pandas' internals and adding new methods do the extension array interface.
+
+String data type
+----------------
+
+Currently, pandas stores text data in an ``object`` -dtype NumPy array.
+The current implementation has two primary drawbacks: First, ``object`` -dtype
+is not specific to strings: any Python object can be stored in an ``object`` -dtype
+array, not just strings. Second: this is not efficient. The NumPy memory model
+isn't especially well-suited to variable width text data.
+
+To solve the first issue, we propose a new extension type for string data. This
+will initially be opt-in, with users explicitly requesting ``dtype="string"``.
+The array backing this string dtype may initially be the current implementation:
+an ``object`` -dtype NumPy array of Python strings.
+
+To solve the second issue (performance), we'll explore alternative in-memory
```

Comment:
actually a much closer soln worth mentioning is to make a DictionaryArray which can back Categorical and StringArray.

Reply:
Jeff, care to answer to what Tom and I replied earlier about this on that not being a general solution for variable strings (not a limited of unique values)? See https://github.com/pandas-dev/pandas/pull/27478#discussion_r306273488

Reply:
Let's keep implementation-related comments in https://github.com/pandas-dev/pandas/issues/8640. If we decide that route is the way to go, we'll update the roadmap to reflect the change of opinion.